ci: e2e tests failing on main

2 E2E tests are currently failing due to a deployment error.

```
Run npm run test:e2e

> @aws/agentcore@0.3.0-preview.6.1 test:e2e
> vitest run --project e2e


 RUN  v4.0.18 /home/runner/work/agentcore-cli/agentcore-cli

 ✓  e2e  e2e-tests/langgraph-anthropic.test.ts > e2e: LangChain_LangGraph/Anthropic — create → deploy → invoke > deploys to AWS successfully 83759ms
 ✓  e2e  e2e-tests/langgraph-gemini.test.ts > e2e: LangChain_LangGraph/Gemini — create → deploy → invoke > deploys to AWS successfully 90601ms
 ✓  e2e  e2e-tests/langgraph-openai.test.ts > e2e: LangChain_LangGraph/OpenAI — create → deploy → invoke > deploys to AWS successfully 90877ms
 ✓  e2e  e2e-tests/langgraph-anthropic.test.ts > e2e: LangChain_LangGraph/Anthropic — create → deploy → invoke > invokes the deployed agent 13255ms
 ✓  e2e  e2e-tests/langgraph-gemini.test.ts > e2e: LangChain_LangGraph/Gemini — create → deploy → invoke > invokes the deployed agent 9130ms
 ✓  e2e  e2e-tests/langgraph-openai.test.ts > e2e: LangChain_LangGraph/OpenAI — create → deploy → invoke > invokes the deployed agent 9706ms
 ✓  e2e  e2e-tests/strands-anthropic.test.ts > e2e: Strands/Anthropic — create → deploy → invoke > deploys to AWS successfully 82924ms
 ✓  e2e  e2e-tests/openaiagents-openai.test.ts > e2e: OpenAIAgents/OpenAI — create → deploy → invoke > deploys to AWS successfully 82096ms
 ✓  e2e  e2e-tests/openaiagents-openai.test.ts > e2e: OpenAIAgents/OpenAI — create → deploy → invoke > invokes the deployed agent 9458ms
 ✓  e2e  e2e-tests/strands-anthropic.test.ts > e2e: Strands/Anthropic — create → deploy → invoke > invokes the deployed agent 13268ms
 ✓  e2e  e2e-tests/googleadk-gemini.test.ts > e2e: GoogleADK/Gemini — create → deploy → invoke > deploys to AWS successfully 107689ms
 ✓  e2e  e2e-tests/googleadk-gemini.test.ts > e2e: GoogleADK/Gemini — create → deploy → invoke > invokes the deployed agent 10469ms
 ✓  e2e  e2e-tests/strands-openai.test.ts > e2e: Strands/OpenAI — create → deploy → invoke > deploys to AWS successfully 80936ms
 ✓  e2e  e2e-tests/strands-gemini.test.ts > e2e: Strands/Gemini — create → deploy → invoke > deploys to AWS successfully 80904ms
 ✓  e2e  e2e-tests/strands-gemini.test.ts > e2e: Strands/Gemini — create → deploy → invoke > invokes the deployed agent 8868ms
 ✓  e2e  e2e-tests/strands-openai.test.ts > e2e: Strands/OpenAI — create → deploy → invoke > invokes the deployed agent 9309ms
stdout | e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > deploys to AWS successfully
Deploy stdout: {"success":false,"error":"CDK deploy failed: ❌  AgentCore-E2eStraBedr39490266-default failed: Error: Entity expansion limit exceeded: 1015 > 1000\n  Deserialization error: to see the raw response, inspect the hidden field {error}.$response on this object.","logPath":"agentcore/.cli/logs/deploy/deploy-20260320-204506.log"}
Deploy stderr: 

 ×  e2e  e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > deploys to AWS successfully 184503ms
   → Deploy failed: : expected 1 to be +0 // Object.is equality
stdout | e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent
Invoke stdout: {"success":false,"error":"No deployed targets found. Run `agentcore deploy` first."}
Invoke stderr: 

stdout | e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent
Invoke stdout: {"success":false,"error":"No deployed targets found. Run `agentcore deploy` first."}
Invoke stderr: 

 ✓  e2e  e2e-tests/strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > deploys to AWS successfully 76980ms
stdout | e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent
Invoke stdout: {"success":false,"error":"No deployed targets found. Run `agentcore deploy` first."}
Invoke stderr: 

stdout | e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke
Teardown stdout: {"success":false,"error":"No resources defined in project. Add at least one resource (agent, memory, evaluator, or gateway) before deploying.","logPath":"agentcore/.cli/logs/deploy/deploy-20260320-204845.log"}
Teardown stderr: 

 ×  e2e  e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent 33546ms
   → Invoke failed: : expected 1 to be +0 // Object.is equality
 ✓  e2e  e2e-tests/langgraph-bedrock.test.ts > e2e: LangChain_LangGraph/Bedrock — create → deploy → invoke > deploys to AWS successfully 86342ms
 ✓  e2e  e2e-tests/strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent 9565ms
 ✓  e2e  e2e-tests/langgraph-bedrock.test.ts > e2e: LangChain_LangGraph/Bedrock — create → deploy → invoke > invokes the deployed agent 8603ms

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL   e2e  e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > deploys to AWS successfully
AssertionError: Deploy failed: : expected 1 to be +0 // Object.is equality

- Expected
+ Received

- 0
+ 1

 ❯ e2e-tests/e2e-helper.ts:136:68
    134|         }
    135| 
    136|         expect(result.exitCode, `Deploy failed: ${result.stderr}`).toB…
       |                                                                    ^
    137| 
    138|         const json = parseJsonOutput(result.stdout) as { success: bool…

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL   e2e  e2e-tests/container-strands-bedrock.test.ts > e2e: Strands/Bedrock — create → deploy → invoke > invokes the deployed agent
AssertionError: Invoke failed: : expected 1 to be +0 // Object.is equality

- Expected
+ Received

- 0
+ 1

 ❯ e2e-tests/e2e-helper.ts:163:72
    161|             }
    162| 
    163|             expect(result.exitCode, `Invoke failed: ${result.stderr}`)…
       |                                                                        ^
    164| 
    165|             const json = parseJsonOutput(result.stdout) as { success: …
 ❯ retry e2e-tests/e2e-helper.ts:30:14
 ❯ e2e-tests/e2e-helper.ts:150:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/2]⎯


 Test Files  1 failed | 10 passed (11)
      Tests  2 failed | 20 passed (22)
   Start at  20:38:27
   Duration  679.32s (transform 278ms, setup 0ms, import 14.43s, tests 1952.74s, environment 2ms)

Error: Process completed with exit code 1.
```

root cause: The fast-xml-parser bump from 5.3.9 → 5.5.7 introduced a new default entity expansion limit of 1000 as part of the CVE-2026-33036/CVE-2026-33349 fix. This causes an issue when CloudFormation responds with XML that contains more expansions than the default limit of 1000 (with 1015). 

This is an issue with a dependency, and we may need to bump it based on https://github.com/aws/aws-sdk-js-v3/issues/7867. 

Acceptance Criteria: 
- [ ] this test is passing on mainline.

--- a/package.json
+++ b/package.json
@@ -132,11 +132,13 @@
   },
   "overridesComments": {
     "minimatch": "GHSA-7r86-cg39-jmmj, GHSA-23c5-xmqv-rm74: minimatch 10.0.0-10.2.2 has ReDoS vulnerabilities. Multiple transitive deps (eslint, typescript-eslint, eslint-plugin-import, eslint-plugin-react, prettier-plugin-sort-imports, aws-cdk-lib) pin older versions. Remove this override once upstream packages update their minimatch dependency to >=10.2.3.",
-    "fast-xml-parser": "GHSA-8gc5-j5rx-235r, GHSA-jp2q-39xq-3w4g: fast-xml-parser <=5.5.6 has entity expansion bypass (CVE-2026-33036, CVE-2026-33349). Transitive via @aws-sdk/xml-builder. Remove once @aws-sdk updates to fast-xml-parser >=5.5.7."
+    "fast-xml-parser": "GHSA-8gc5-j5rx-235r, GHSA-jp2q-39xq-3w4g: fast-xml-parser <=5.5.6 has entity expansion bypass (CVE-2026-33036, CVE-2026-33349). Transitive via @aws-sdk/xml-builder. Remove once @aws-sdk updates to fast-xml-parser >=5.5.7.",
+    "@aws-sdk/xml-builder": "aws/aws-sdk-js-v3#7867: @aws-sdk/xml-builder <3.972.14 does not configure maxTotalExpansions on fast-xml-parser, causing 'Entity expansion limit exceeded' on large CloudFormation responses. Remove once @aws-sdk/client-* deps are bumped past 3.972.14."
   },
   "overrides": {
     "minimatch": "10.2.4",
-    "fast-xml-parser": "5.5.7"
+    "fast-xml-parser": "5.5.7",
+    "@aws-sdk/xml-builder": "3.972.15"
   },
   "engines": {
     "node": ">=20"

